tests/Doctrine/Tests/DBAL/Functional/ConnectionTest.php

```diff
@@ -402,10 +406,23 @@ public function testResultCompatibilityWhenExecutingCacheQuery(): void
             $this->connection->getDatabasePlatform()->getDummySelectSQL(),
             [],
             [],
-            new QueryCacheProfile(1, 'cacheKey', new ArrayCache())
+            new QueryCacheProfile(1, 'cacheKey', $this->getArrayCache())
         );
 
         self::assertInstanceOf(Result::class, $result);
         self::assertInstanceOf(Driver\ResultStatement::class, $result);
     }
+
+    private function getArrayCache(): Cache
+    {
+        if (class_exists(DoctrineProvider::class)) {
+            return DoctrineProvider::wrap(new ArrayAdapter());
+        }
+
+        if (class_exists(ArrayCache::class)) {
+            return new ArrayCache();
+        }
+
+        self::markTestSkipped();
```

Comment:
As a second thought, I think failing (`self::fail('Cannot instantiate cache backend.')`) would be a safer option (like it would fail in the original implementation).

Reply:
Sure.